src/connector/codec/src/decoder/avro/mod.rs

```diff
@@ -405,13 +401,23 @@ impl Access for AvroAccess<'_> {
                     // },
                     // ...]
                     value = v;
-                    options.schema = options.extract_inner_schema(None);
+                    let Schema::Union(u) = options.schema else {
+                        return Err(create_error());
+                    };
+                    let Some(schema) = get_nullable_union_inner(u) else {
+                        return Err(create_error());
+                    };
+                    options.schema = schema;
                     continue;
                 }
                 Value::Record(fields) => {
-                    if let Some((_, v)) = fields.iter().find(|(k, _)| k == key) {
-                        value = v;
-                        options.schema = options.extract_inner_schema(Some(key));
+                    let Schema::Record(record_schema) = &options.schema else {
+                        return Err(create_error());
+                    };
+                    if let Some(idx) = record_schema.lookup.get(key) {
+                        // if let Some((_, v)) = fields.iter().find(|(k, _)| k == key) {
```

Comment:
why to leave this comment? and what if lookup failed? 🤔

Reply:
* When I cleanup I only checked `convert_to_datum` but missed `access`
* When lookup fails, the behavior is as before: `continue` is not executed so `Err(create_error())?;` at the end of `while`-body is executed. This function returns an error to the caller and the caller would set this whole column as `NULL`.